### PR TITLE
Enable / disable buttons when back end is down

### PR DESCRIPTION
**Pull request by a-breid-ericsson**
* Buttons that require back end to be up and running will now be disabled
  when back end is down.
* A status text will be displayed when buttons are disabled hinting the user
  to switch back end or check if it is up.

### Sign-off

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

Signed-off-by:

### CODE REVIEW

**Comment by a-breid-ericsson** [comment]
Test rules enabling/disabling may be another story.

**Comment by e-edling-ericsson** [comment]
Well, if front-end cannot connect there's no reason for the buttons to work anyways?

Although after more thought, i still think that they always should be visible. We got the button "EI Backend Status" to know if the backend is down or not. I don't really see the use case with the buttons to not show up if the backend is down(the user will know about that anyways).

**Comment by a-breid-ericsson** [comment]
@e-edling-ericsson  The buttons are displayed always. Try it out.

**Comment by tobiasake** [comment]
Ok.
I meant all situations, including when Ei backend webserver responds with different errors and if connections is blocked between Frontend and backend, or what the issues can be.
But as long as its works stable, go for it.

File: src/main/resources/templates/subscription.html [hunk 1]
```diff
@@ -15,11 +15,16 @@
 
 <body>
     <div class="d-none" id="subscriptionTemplate" th:text="${subscriptionTemplate}"></div>
-    <div class="d-flex">
-        <div class="col-md-9 col-8">
+    <div class="d-flex flex-wrap">
+        <div class="col-md-4 col-6">
```
Comment (Christoffer-Cortes) [inline src/main/resources/templates/subscription.html]: order-md-0, is missing in class.

File: src/main/resources/templates/subscription.html [hunk 2]
```diff
@@ -15,11 +15,16 @@
...
             <h3>Eiffel Intelligence Subscription Handling</h3>
         </div>
-        <div class="col-md-3 col-4 d-flex justify-content-end">
+        <div class="col-md-6 col-12">
```
Comment (Christoffer-Cortes) [inline src/main/resources/templates/subscription.html]: order-last order-md-1, is missing in class.

File: src/main/resources/templates/subscription.html [hunk 3]
```diff
@@ -15,11 +15,16 @@
...
+            <span id="back_end_down_warning" class="hidden_by_default">
+                Ensure back end is up or switch to another back end before continuing!
+            </span>
+        </div>
+        <div class="col-md-2 col-6 d-flex justify-content-end">
```
Comment (Christoffer-Cortes) [inline src/main/resources/templates/subscription.html]: order-md-2, is missing in class.

File: src/main/resources/static/js/subscription.js [hunk 4]
```diff
@@ -92,6 +92,23 @@ jQuery(document).ready(function () {
     // Check if EI Backend Server is online every X seconds
     window.setInterval(function () { checkBackendStatus(); }, 15000);
 
+    // Check if buttons should be enabled or disabled
+    // Execute code to show or hide status enable disable buttons
+    var buttonsDiabled = false;
```
Comment (Christoffer-Cortes) [inline src/main/resources/static/js/subscription.js]: buttonsDisabled is misspelled.

Comment (a-breid-ericsson) [reply src/main/resources/static/js/subscription.js]: Done

File: src/main/resources/templates/subscription.html (hunk 3 above)
Comment (a-breid-ericsson) [reply src/main/resources/templates/subscription.html]: Done

File: src/main/resources/templates/subscription.html (hunk 2 above)
Comment (a-breid-ericsson) [reply src/main/resources/templates/subscription.html]: Done

File: src/main/resources/templates/subscription.html (hunk 1 above)
Comment (a-breid-ericsson) [reply src/main/resources/templates/subscription.html]: Done

**Pull request merged**